src/main/java/com/gluonhq/gradle/attach/AttachConfiguration.java

```diff
@@ -31,23 +31,25 @@
  */
 package com.gluonhq.gradle.attach;
 
-import com.gluonhq.gradle.ClientExtension;
-import com.gluonhq.omega.attach.AttachResolver;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
 import org.gradle.api.Action;
 import org.gradle.api.NamedDomainObjectContainer;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.Configuration;
 
-import javax.inject.Inject;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
+import com.gluonhq.gradle.ClientExtension;
 
 public class AttachConfiguration {
+	private static final String DEPENDENCY_GROUP = "com.gluonhq.attach";
+	private static final String UTIL_ARTIFACT = "util";
 
     private Project project;
-
-    private String version = "4.0.2";
+    private String version = "4.0.3";
```

Comment:
I have pushed an attempted rewrite, as I parse the maven plugin changes.
But it fails, since the version field is always null. Where is the version supposed to come from?

```
Found module name 'hellogluon'                                 
Adding dependency for display in configuration implementation: {name=display, classifier=desktop, version=null, group=com.gluonhq.attach}
Adding dependency for lifecycle in configuration implementation: {name=lifecycle, classifier=desktop, version=null, group=com.gluonhq.attach} 
Adding dependency for statusbar in configuration implementation: {name=statusbar, classifier=, version=null, group=com.gluonhq.attach}
Adding dependency for storage in configuration implementation: {name=storage, classifier=desktop, version=null, group=com.gluonhq.attach}
```

Any idea of what I am missing?

Reply:
This is one of the main differences between Gradle and Maven...

In Maven, you _have to_ define your dependencies in the pom, therefore you add all Attach dependencies, including their version, and that's why the plugin can take if from the dependencies and there is no need to set a default value.

In Gradle, while you can also add a list of dependencies, we could avoid it and do the dependency "injection", by taking the list of Attach services from the clientPlugin definition, and using the version, if defined, or a default version that was hardcoded in the plugin itself.

I'd rather remove that default version from the plugin, so that would mean that the `version` field in the client plugin will be mandatory. 

Does this make sense to you?

Reply:
Makes perfect sense. I will make it so (later today).

Reply:
Done. Throws exception if version is not specified.